libdocker/kube_docker_client.go

```diff
@@ -314,21 +314,21 @@ func (p *progress) set(msg *dockermessage.JSONMessage) {
 	p.timestamp = time.Now()
 }
 
-func (p *progress) get() (string, time.Time) {
+func (p *progress) get() (status, msg string, timestamp time.Time) {
 	p.RLock()
 	defer p.RUnlock()
 	if p.message == nil {
-		return "No progress", p.timestamp
+		return "", "No progress", p.timestamp
 	}
 	// The following code is based on JSONMessage.Display
 	var prefix string
 	if p.message.ID != "" {
 		prefix = fmt.Sprintf("%s: ", p.message.ID)
 	}
 	if p.message.Progress == nil {
-		return fmt.Sprintf("%s%s", prefix, p.message.Status), p.timestamp
+		return p.message.Status, fmt.Sprintf("%s%s", prefix, p.message.Status), p.timestamp
```

Comment:
Is there a real need to return status and msg? They are the same in many occasions

Reply:
Yes, `msg` includes `status`.  `status` is used for matching `Extracting`.

Reply:
Seems redundant to me

Reply:
I'm not thrilled with it, either. It would be much cleaner to factor out the formatting of the message into a separate function, and have `(*progress).get()` return the JSONMessage pointer directly.
```go
func (*progress) get() (*dockermessage.JSONMessage, time.Time)

func formatProgress(*dockermessage.JSONMessage) string
```

Reply:
Thanks for the advice. Resolved in new commit [44b300b](https://github.com/Mirantis/cri-dockerd/pull/376/commits/44b300bfbea68e72cddc46224e64ceb553795217)